### ISSUE TITLE
Fixture Indexes' name problem. 

### ISSUE THREAD

**Issue by jorgeandrescastro** [cakephp/cakephp#7608]
Hello,
We have stumbled upon the following warning message while running our test cases on our Cakephp 3.1.2 project:
```
Warning Error: Fixture creation for "threads" failed "SQLSTATE[HY000]: General error: 1 index user_id already exists" in [/var/www/project/vendor/cakephp/cakephp/src/TestSuite/Fixture/TestFixture.php, line 240]
```

Our tests are currently using two tables that have `belongsTo` associations with our users table, thus when we bake our fixtures, both tables fixtures define the index name for the user_id fields as `user_id`:
```php
'_indexes' => 
	[
        'user_id' => ['type' => 'index', 'columns' => ['user_id'], 'length' => []],
            
    ],
```
After some research, we have found that tables' indexes names should be unique, so we believe this is the reason of the warning. 

We are wondering if we might be making a mistake in our fixture baking?
```
bin/cake bake fixture Threads -n 2
```
 or if it perhaps this is a normal behavior of the fixture baking process, and that by default the names of the index for fixtures is the name of the field itself?

I thank you in advance for your response; should you need more information, please let me know. 
Thanks

**Comment by markstory** [comment]
I can take a look.

**Comment by markstory** [comment]
I'm assuming you are using MySQL. What does `SHOW CREATE TABLE threads` output? On my setup, constraints, and indexes used names defined in the schema. For example:

```sql
CREATE TABLE `bookmarks` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `user_id` int(11) NOT NULL,
  `title` varchar(50) DEFAULT NULL,
  `description` text,
  `url` text,
  `created` datetime DEFAULT NULL,
  `updated` datetime DEFAULT NULL,
  `starred` tinyint(1) DEFAULT '0',
  PRIMARY KEY (`id`),
  KEY `user_key` (`user_id`),
  CONSTRAINT `bookmarks_ibfk_1` FOREIGN KEY (`user_id`) REFERENCES `users` (`id`)
)
```

Generated the following fixture schema with `bin/cake bake Bookmarks`:

```php
class BookmarksFixture extends TestFixture
{

    public $fields = [
        'id' => ['type' => 'integer', 'length' => 11, 'unsigned' => false, 'null' => false, 'default' => null, 'comment' => '', 'autoIncrement' => true, 'precision' => null],
        'user_id' => ['type' => 'integer', 'length' => 11, 'unsigned' => false, 'null' => false, 'default' => null, 'comment' => '', 'precision' => null, 'autoIncrement' => null],
        'title' => ['type' => 'string', 'length' => 50, 'null' => true, 'default' => null, 'comment' => '', 'precision' => null, 'fixed' => null],
        'description' => ['type' => 'text', 'length' => null, 'null' => true, 'default' => null, 'comment' => '', 'precision' => null],
        'url' => ['type' => 'text', 'length' => null, 'null' => true, 'default' => null, 'comment' => '', 'precision' => null],
        'created' => ['type' => 'datetime', 'length' => null, 'null' => true, 'default' => null, 'comment' => '', 'precision' => null],
        'updated' => ['type' => 'datetime', 'length' => null, 'null' => true, 'default' => null, 'comment' => '', 'precision' => null],
        'starred' => ['type' => 'boolean', 'length' => null, 'null' => true, 'default' => '0', 'comment' => '', 'precision' => null],
        '_indexes' => [
            'user_key' => ['type' => 'index', 'columns' => ['user_id'], 'length' => []],
        ],
        '_constraints' => [
            'primary' => ['type' => 'primary', 'columns' => ['id'], 'length' => []],
            'bookmarks_ibfk_1' => ['type' => 'foreign', 'columns' => ['user_id'], 'references' => ['users', 'id'], 'update' => 'restrict', 'delete' => 'restrict', 'length' => []],
        ],
        '_options' => [
            'engine' => 'InnoDB',
            'collation' => 'utf8_general_ci'
        ],
    ];
```

**Comment by jorgeandrescastro** [comment]
Hi @markstory thanks for taking a look at this.
And yes, you are right, the fixture baking will take the name of the index, not the name of the field as I had assumed. The problem I was having is that we create our tables with migrations, using:
```php
->addForeignKey('user_id', 'users', 'id', ['delete' => 'CASCADE', 'update' => 'CASCADE'])
```
and this will automatically create an index in the table, using the field name, in this case, `user_id`.
I tested creating the table using also the `addIndex` function:
```php
->addIndex(['user_id'], ['name' => 'user_key'])
```
and then it worked as expected, creating the index with the name I had specified. 
Thanks for clarifying that out for me. :)
